src/NUnitEngine/nunit.engine/Services/ExtensionService.cs

```diff
@@ -537,7 +517,37 @@ internal void FindExtensionsInAssembly(ExtensionAssembly assembly)
             }
         }
 
-#endregion
+        /// <summary>
+        /// Checks that the target framework of the current runner can load the extension assembly. For example, .NET Core
+        /// cannot load .NET Framework assemblies and visa-versa.
+        /// </summary>
+        /// <param name="runnerAsm">The executing runner</param>
+        /// <param name="extensionAsm">The extension we are attempting to load</param>
+        internal static void ValidateTargetFramework(Assembly runnerAsm, ExtensionAssembly extensionAsm)
+        {
+            if (runnerAsm == null)
+                return;
+
+            var extHelper = new TargetFrameworkHelper(extensionAsm.FilePath);
+            var runnerHelper = new TargetFrameworkHelper(runnerAsm.Location);
+            if (runnerHelper.FrameworkName?.StartsWith(".NETStandard") == true)
+            {
+                throw new NUnitEngineException("Test runners must target .NET Core or .NET Framework, not .NET Standard");
+            }
+            else if (runnerHelper.FrameworkName?.StartsWith(".NETCoreApp") == true)
+            {
+                if (extHelper.FrameworkName?.StartsWith(".NETStandard") != true && extHelper.FrameworkName?.StartsWith(".NETCoreApp") != true)
+                {
+                    throw new NUnitEngineException(".NET Core runners require .NET Core or .NET Standard extensions");
+                }
+            }
+            else if (extHelper.FrameworkName?.StartsWith(".NETCoreApp") == true || extHelper.FrameworkName?.StartsWith(".NETStandard") == true)
+            {
+                throw new NUnitEngineException(".NET Framework runners cannot load .NET Core extensions");
```

Comment:
This exception should maybe say .NET Core/.NET Standard extensions. I'd expect .NET Core extensions not to be loadable, it's .NET Standard extensions that are the 'unusual case' here in my opinion. 🙂